HHS/Head-Start-TTADP
# [TTAHUB-3530] Activity Report FE changes for monitoring

## Description of change

This hooks up the monitoring goal to the standard goal drop down and adds the citations drop down to the objective.

Note: This does NOT include the various alerts on the FE for Grants under monitoring missing monitoring goals that's a later ticket.

## How to test

In order to setup the goals and the retrieval of Citations you will need to complete the following steps.

- Restore a prod db to have access to the monitoring data (run db:migrate)
- Open createMonitoringGoals.js and set the cutOffDate var to a date in the past like '2024-10-01'
- Run the Create Monitor Goal job via the CLI as _yarn docker:yarn:be createMonitoringGoalsCLI:local_ (or the local way)
- Ensure you now have monitoring goals in the goals table.
- Open src\services\Citation.js and make sure the cutoff date var coresponds with what you set in the job ( '2024-10-01').
- Find a recipient that is linked to a monitoring grant using the below query (from @thewatermethod).
- Create an AR, be sure to select the monitoring grant AND set a start date before going to the goals and objectives page.
- Check the 'Use standard goals' check box on the goals and objectives page and search for the goal that starts with "(Monitoring)".
- When you add a objective to that goal you should now have the ability to select citaions.
- Test various combinations of Grants that require monitoring with ones that dont. Make sure we only create monitoring goals accordingly.
- Test Creating and updating the goal fields, reviewing the report, submitting, etc.


## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-0


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
